src/sentry/grouptype/grouptype.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+
+from sentry.event_manager import DEFAULT_GROUPHASH_IGNORE_LIMIT
+from sentry.types.issues import GroupCategory
+
+_group_type_registry = {}
+
+
+@dataclass(frozen=True)
+class GroupType:
+    type_id: int
+    slug: str
+    description: str
+    category: int
+    ignore_limit: int = DEFAULT_GROUPHASH_IGNORE_LIMIT
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        if _group_type_registry.get(cls.type_id):
+            raise ValueError(
+                f"A group type with the type_id {cls.type_id} has already been registered."
+            )
+        _group_type_registry[cls.type_id] = cls
+
+    def __post_init__(self):
+        valid_categories = [category.value for category in GroupCategory]
+        if self.category not in valid_categories:
+            raise ValueError(f"Category must be one of {valid_categories} from GroupCategory.")
+
+
+def get_group_types_by_category(category):
+    return [child.type_id for child in GroupType.__subclasses__() if child.category == category]
+
+
+def get_group_type_by_slug(slug):
+    for group_type in _group_type_registry.values():
+        if group_type.slug == slug:
+            return group_type
+    raise ValueError(f"No group type with the slug {slug} is registered.")
+
+
+@dataclass(frozen=True)
+class ErrorGroupType(GroupType):
+    type_id = 1
+    slug = "error"
+    description = "Error"
+    category = GroupCategory.ERROR.value
+    ignore_limit = 0
+
+
+@dataclass(frozen=True)
+class SlowDBQueryGroupType(GroupType):
+    type_id = 1001
+    slug = "performance_slow_db_query"
+    description = "Slow DB Query"
+    category = GroupCategory.PERFORMANCE.value
+    ignore_limit = 100
+
+
+@dataclass(frozen=True)
+class PerformanceRenderBlockingAssetSpanGroupType(GroupType):
+    type_id = 1004
+    slug = "performance_render_blocking_asset_span"
+    description = "Large Render Blocking Asset"
+    category = GroupCategory.PERFORMANCE.value
+
+
+@dataclass(frozen=True)
+class PerformanceNPlusOneGroupType(GroupType):
+    type_id = 1006
+    slug = "performance_n_plus_one_db_queries"
+    description = "N+1 Query"
+    category = GroupCategory.PERFORMANCE.value
+
+
+@dataclass(frozen=True)
+class PerformanceConsecutiveDBQueriesGroupType(GroupType):
+    type_id = 1007
+    slug = "performance_consecutive_db_queries"
+    description = "Consecutive DB Queries"
+    category = GroupCategory.PERFORMANCE.value
```

Comment:
https://github.com/getsentry/sentry/pull/43919
think we have a new `ignore_limit` here coming down the pipeline

Reply:
Great catch, I'll sneak that change in too